sdk/main/src/Transaction.cc

```diff
@@ -128,28 +124,92 @@ proto::TransactionBody Transaction<SdkRequestType>::generateTransactionBody(cons
   body.set_transactionfee(static_cast<uint64_t>(getMaxTransactionFee(client).toTinybars()));
   body.set_allocated_memo(new std::string(mTransactionMemo));
   body.set_allocated_transactionvalidduration(internal::DurationConverter::toProtobuf(mTransactionValidDuration));
-  body.set_allocated_nodeaccountid(mNodeAccountId->toProtobuf().release());
+  body.set_allocated_nodeaccountid(mNodeAccountId.toProtobuf().release());
   return body;
 }
 
 //-----
 template<typename SdkRequestType>
 Hbar Transaction<SdkRequestType>::getMaxTransactionFee(const Client& client) const
 {
-  if (mMaxTransactionFee != mDefaultMaxTransactionFee)
-  {
-    return mMaxTransactionFee;
-  }
+  return mMaxTransactionFee              ? *mMaxTransactionFee
+         : client.getMaxTransactionFee() ? *client.getMaxTransactionFee()
+                                         : DEFAULT_MAX_TRANSACTION_FEE;
+}
+
+//-----
+template<typename SdkRequestType>
+TransactionResponse Transaction<SdkRequestType>::mapResponse(const proto::TransactionResponse& response) const
+{
+  TransactionResponse txResp = TransactionResponse::fromProtobuf(response);
+  txResp.mTransactionId = mTransactionId;
+  return txResp;
+}
+
+//-----
+template<typename SdkRequestType>
+Status Transaction<SdkRequestType>::mapResponseStatus(const proto::TransactionResponse& response) const
+{
+  return STATUS_MAP.at(response.nodetransactionprecheckcode());
+}
 
-  else if (const std::unique_ptr<Hbar> defaultMaxTxFee = client.getDefaultMaxTransactionFee())
+//-----
+template<typename SdkRequestType>
+typename Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
+  ExecutionStatus
+  Transaction<SdkRequestType>::shouldRetry(Status status,
+                                           const Client& client,
+                                           const proto::TransactionResponse& response)
+{
+  if (status == Status::TRANSACTION_EXPIRED)
```

Comment:
This deeply nested structure is *tricky* to understand. What do you think of this simplified form?

```
  // If the Transaction's not expired, forward to the base class.
  if (status != Status::TRANSACTION_EXPIRED)
  {
    return Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::shouldRetry(
      status, client, response);
  }

  // return an error if explicitly configured to do so
  const std::optional<bool> clientTxIdRegenPolicy = client.getTransactionIdRegenerationPolicy();
  if (mTransactionIdRegenerationPolicy && !*mTransactionIdRegenerationPolicy ||
      clientTxIdRegenPolicy && !*clientTxIdRegenPolicy)
  {
    return Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
      ExecutionStatus::REQUEST_ERROR;
  }

  // otherwise, regenerate transaction IDs by default
  mTransactionId = TransactionId::generate(mTransactionId.getAccountId());
  return Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
    ExecutionStatus::RETRY;
```

Reply:
It is tricky but necessary. Your proposed form would not work in the case that the `Transaction` was explicitly-set to regenerate transaction IDs and the `Client` was explicitly-set to not regenerate transaction IDs; it would still return a `REQUEST_ERROR` even though the `Transaction` behavior should override the `Client` behavior. I did try to condense it a little bit however.

Reply:
With that in mind, what about the following:

```
  // If the Transaction's not expired, forward to the base class.
  if (status != Status::TRANSACTION_EXPIRED)
  {
    return Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
      determineStatus(status, client, response);
  }

  bool retry = false;

  // Follow this Transaction's policy if it has been explicitly set
  if (mTransactionIdRegenerationPolicy)
  {
    retry = *mTransactionIdRegenerationPolicy;
  }

  // Follow the Client's policy if this Transaction's policy hasn't been explicitly set and the Client's policy has been
  else if (const std::optional<bool> clientTxIdRegenPolicy = client.getTransactionIdRegenerationPolicy();
           clientTxIdRegenPolicy)
  {
    retry = *clientTxIdRegenPolicy;
  }

  // If no policy has been explicitly-set, regenerate transaction IDs by default
  else
  {
    retry = true;
  }

  if (retry)
  {
    mTransactionId = TransactionId::generate(mTransactionId.getAccountId());
    return Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
      ExecutionStatus::RETRY;
  }

  // Return REQUEST_ERROR if the transaction expired but transaction IDs aren't allowed to be regenerated
  return Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
    ExecutionStatus::REQUEST_ERROR;
```

Reply:
(or you could even default `retry` to true, and get rid of the `else` block)

Reply:
Updated